src/App.vue

```diff
@@ -291,6 +291,33 @@ export default class App extends Vue {
       this.toggleRightPanel(null);
     }
   }
+
+  async goToNewDataset() {
+    try {
+      // Ensure we have a valid user and location before proceeding
+      if (!this.store.girderUser || !this.store.isLoggedIn) {
+        throw new Error("User must be logged in");
+      }
+
+      // Get the user's private folder
+      const privateFolder = await this.store.api.getUserPrivateFolder();
```

Comment:
Calling an API in a button action is fine, but it might delay the UI reaction if the server is loaded.   
The user might think that they didn't click and will click again, calling the API again and loading the server even more.  
To fix this, you can:
 - Add a loader by using the `loading` prop of `v-btn`
 - Or get the privateFolder of the user when they connect and store it in the store